src/core/lib/Modhex.mjs

```diff
@@ -0,0 +1,174 @@
+/**
+ * Modhex functions.
+ *
+ * @author linuxgemini [ilteris@asenkron.com.tr]
+ * @copyright Crown Copyright 2020
+ * @license Apache-2.0
+ */
+
+import Utils from "../Utils.mjs";
+import { toHex, toHexFast, fromHex } from "./Hex.mjs";
+
+/**
+ * Convert a byte array into a modhex string.
+ *
+ * @param {byteArray|Uint8Array|ArrayBuffer} data
+ * @param {string} [delim=" "]
+ * @param {number} [padding=2]
+ * @returns {string}
+ *
+ * @example
+ * // returns "cl bf bu"
+ * toModhex([10,20,30]);
+ *
+ * // returns "cl:bf:bu"
+ * toModhex([10,20,30], ":");
+ */
+export function toModhex(data, delim=" ", padding=2, extraDelim="", lineSize=0) {
```

Comment:
`extraDelim` and `lineSize` are used in this function but untested.

Reply:
This is passed along to the `toHex` function, I think this can be removed.

Reply:
Looked at the code and it seems that `lineSize` is in fact used.